packages/components/inputs/localized-money-input/src/localized-money-input.tsx

```diff
@@ -391,23 +391,26 @@ LocalizedMoneyInput.defaultProps = {
   horizontalConstraint: 'scale',
 };
 
-LocalizedMoneyInput.convertToMoneyValues = (values: TValue[], locale: string) =>
-  Object.values(values).map((value) =>
-    MoneyInput.convertToMoneyValue(value, locale)
+LocalizedMoneyInput.convertToMoneyValues = (
+  values: TValue[],
+  locale: string
+): TMoneyValue[] =>
+  Object.values(values).map(
+    (value) => MoneyInput.convertToMoneyValue(value, locale) as TMoneyValue
```

Comment:
I would prefer to avoid casting unless absolutely necessary.

For instance, functions like `map`, `reduce`, etc. they all accept type arguments, so that you shouldn't need to type things explicitly.

For example:

```ts
map<TMoneyValue>()

reduce<Record<TCurrencyCode, TValue>>()
```

Can you try that please?

Reply:
What I noticed here after giving it a look again is that the map function possibly returns a null (strange) as one of its return types asides the `<TMoneyValue>`. I have tried `map<TMoneyValue>` and `map<TMoneyValue | null>`, both do not seem to allow except casting the mapped return type as `map<TMoneyValue>`

Reply:
I'll take a look

Reply:
Thank you